avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeBase.java

```diff
@@ -109,6 +109,13 @@ protected Class compileClass(final String className, Set<String> knownUsedFullyQ
 
     JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
     String compileClassPathForCurrentFile = Utils.inferCompileDependencies(compileClassPath, filePath, knownUsedFullyQualifiedClassNameSet);
+    if (null == compiler) {
+      /**
+       * If the above function returns null, it is very likely that the env setting: "JAVA_HOME" is not being setup properly.
+       */
+      throw new FastSerdeGeneratorException("Couldn't locate java compiler at runtime, please double check your env "
```

Comment:
i'd change the wording a bit: "Runtime compiler not found. are we running under a JRE and not a JDK? JAVA_HOME=x, java.home=y, PATH=z"